scripts/uberenv/packages/mfem/package.py

```diff
@@ -44,6 +44,7 @@ class Mfem(Package):
     # SERAC EDIT BEGIN
     # Version spec used in serac based on mfem@develop commit SHA
     version('4.1.0p1', commit='bb3c788a05f430bcfed03065ba39868974964924')
+    version('4.2', tag='v4.2')
```

Comment:
We should pull in the official tarball version of 4.2.0 from their spack package. These package updates should also probably get incorporated: https://github.com/spack/spack/pull/19748 .

Reply:
The Spack changes have been merged in, and once this PR is merged I can open a PR to add the AMGX integration to Spack (which isn't part of that PR)